onnx/defs/tensor/defs.cc

```diff
@@ -2663,6 +2663,14 @@ ONNX_OPERATOR_SET_SCHEMA(
             "Constrain to all tensor types.")
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
           updateOutputElemType(ctx, 0, TensorProto::INT64);
+          auto* output_shape = ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+          auto& input_shape = getInputShape(ctx, 0);
+          int64_t rank = input_shape.dim_size();
+          // NonZero output is of rank 2.
+          // First dimension is same as input rank
+          output_shape->add_dim()->set_dim_value(rank);
```

Comment:
I think that we should handle the case where the input rank is not known. Check ```if (hasInputShape(ctx,0))``` before calling ```getInputShape(ctx,0)```

Reply:
I made the suggested change. If someone could take a look, it would be helpful, thanks!